Description: Make FAT UUID uppercase to match Linux
Author: Vladimir Serbinenko <phcoder@gmail.com>
Origin: upstream, http://bazaar.launchpad.net/~vcs-imports/grub/grub2-bzr/revision/3482
Bug-Ubuntu: https://bugs.launchpad.net/bugs/948716
Forwarded: not-needed
Applied-Upstream: http://bazaar.launchpad.net/~vcs-imports/grub/grub2-bzr/revision/3482
Last-Update: 2012-03-19

Index: b/grub-core/fs/fat.c
===================================================================
--- a/grub-core/fs/fat.c
+++ b/grub-core/fs/fat.c
@@ -836,9 +836,12 @@
   data = grub_fat_mount (disk);
   if (data)
     {
+      char *ptr;
       *uuid = grub_xasprintf ("%04x-%04x",
 			     (grub_uint16_t) (data->uuid >> 16),
 			     (grub_uint16_t) data->uuid);
+      for (ptr = *uuid; ptr && *ptr; ptr++)
+	*ptr = grub_toupper (*ptr);
     }
   else
     *uuid = NULL;
